### PR TITLE
feat: conway primitives

### PR DESCRIPTION
**Pull request by jmhrpr**
WIP, not tested

### CODE REVIEW

**Comment by scarmuega** [comment]
wow! amazing 🚀 

do you know if any Babbage tx can be considered a subset of Conway? or is anything in Conway that breaks backward compatibility?

I'm asking because we could potentially introduce a `ConwayCompatible` concept as we do with `AlonzoCompatible` to reduce the total number of distinct branches.

**Comment by jmhrpr** [comment]
> wow! amazing 🚀
> 
> do you know if any Babbage tx can be considered a subset of Conway? or is anything in Conway that breaks backward compatibility?
> 
> I'm asking because we could potentially introduce a `ConwayCompatible` concept as we do with `AlonzoCompatible` to reduce the total number of distinct branches.

A field was removed from transaction body (update field) and multiple were added, I don’t know if that answers your question or not. Also a number of types within the transaction body were changed.

**Comment by scarmuega** [comment]
yes, that answers my question.

If Conway removes the `update` field then it breaks compatibility and we can't use the same primitives.

PR looks good as it is then. Anything missing besides some tests?

**Comment by jmhrpr** [comment]
> yes, that answers my question.
> 
> If Conway removes the `update` field then it breaks compatibility and we can't use the same primitives.
> 
> PR looks good as it is then. Anything missing besides some tests?

There is pretty much no data on sanchonet to test against at the moment, so I plan on building some transactions with the CLI at some point. But it successfully decoded all the Conway blocks that were present on the chain a couple days ago.

**Comment by scarmuega** [comment]
Makes sense. I say we merge it as it is and add a follow-up issue to include some tests once there's more data available.

Feel free to flag the PR ready for review if you agree.

**Comment by jmhrpr** [comment]
I will give it a review against the CDDL at some point soon to try catch any mistakes first, then mark ready for review

File: pallas-traverse/src/era.rs [hunk 1]
```diff
@@ -13,6 +13,7 @@ impl Era {
             Feature::CIP31 => self.ge(&Era::Babbage),
             Feature::CIP32 => self.ge(&Era::Babbage),
             Feature::CIP33 => self.ge(&Era::Babbage),
+            // TODO: Conway features
```
Comment (scarmuega) [inline pallas-traverse/src/era.rs:16]: I would create a new feature called `Governance` or `CIP1694`

**Comment by kevinhammond** [comment]
> wow! amazing 🚀
> 
> do you know if any Babbage tx can be considered a subset of Conway? or is anything in Conway that breaks backward compatibility?

Governance proposals and MIRs are removed from Conway (superseded).  I think all the other changes will be backwards compatible (CIP1694, Plutus v3 etc)

**Comment by scarmuega** [comment]
thank you @kevinhammond, that confirms our findings.

@jmhrpr should we merge using the `unstable` flag as we discussed and continue with the remaining tasks in a different PR?

**Comment by lisicky** [comment]
Hi guys. PR looks good, but I have changes some things since CDDL was changed. And found that some of plutus related struct didn't implement plutus v3 . All of these changes I pushed into the [PR](https://github.com/jmhrpr/pallas/pull/3)

```
CDDL changes:
- NewCommittee was renamed to UpdateCommittee
- Committee was deleted

Update plutus v3 related structs
I also added a test with manually constructed block with a tx which contains all conway certs, voting proposals and voting procedures
```

But also I have a question, did you guys have ever tested deserialisation of  babbage block with plutus v2 in auxiliary data ?

**Comment by lisicky** [comment]
Also we wanna to use oura with conway support. Is it possible to merge this changes ?

**Comment by scarmuega** [comment]
@jmhrpr if you can fix the merge conflicts, I'll merge it as it is and apply the `unstable` flag that we discussed.

**Comment by jmhrpr** [comment]
@scarmuega done

**Review by scarmuega — APPROVED** [review]

**Pull request merged**